src/mds/Server.cc

```diff
@@ -480,7 +480,7 @@ void Server::handle_client_session(const cref_t<MClientSession> &m)
   dout(3) << "handle_client_session " << *m << " from " << m->get_source() << dendl;
   ceph_assert(m->get_source().is_client()); // should _not_ come from an mds!
 
-  if (!session) {
+  if (!session || !session->is_open()) {
```

Comment:
Get this failure:

````
2019-11-12T08:23:14.232 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi154.stderr:2019-11-12T08:23:14.235+0000 7f0eb2b37e00 -1 init, newargv = 0x55cf51d09260 newargc=7
2019-11-12T08:23:14.233 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi154.stderr:ceph-fuse[17353]: starting ceph client
2019-11-12T08:23:14.236 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi154.stderr:2019-11-12T08:23:14.238+0000 7f0e9affd700 -1 client.4618 mds.0 rejected us (sessionless)
2019-11-12T08:23:14.236 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi154.stderr:ceph-fuse[17353]: ceph mount failed with (1) Operation not permitted
2019-11-12T08:23:14.244 DEBUG:teuthology.orchestra.run:got remote process result: 2
2019-11-12T08:23:14.255 INFO:teuthology.orchestra.run.smithi154.stderr:ls: cannot access /sys/fs/fuse/connections: No such file or directory
2019-11-12T08:23:14.402 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi154.stderr:daemon-helper: command failed with exit status 1
2019-11-12T08:23:15.357 INFO:teuthology.orchestra.run:Running command with timeout 900
2019-11-12T08:23:15.357 INFO:teuthology.orchestra.run.smithi154:Running:
2019-11-12T08:23:15.358 INFO:teuthology.orchestra.run.smithi154:> sudo mount -t fusectl /sys/fs/fuse/connections /sys/fs/fuse/connections
2019-11-12T08:23:15.477 DEBUG:teuthology.orchestra.run:got remote process result: 32
2019-11-12T08:23:15.478 INFO:teuthology.orchestra.run.smithi154.stderr:mount: mount point /sys/fs/fuse/connections does not exist
2019-11-12T08:23:15.479 INFO:teuthology.orchestra.run:Running command with timeout 900
2019-11-12T08:23:15.479 INFO:teuthology.orchestra.run.smithi154:Running:
2019-11-12T08:23:15.479 INFO:teuthology.orchestra.run.smithi154:> ls /sys/fs/fuse/connections
2019-11-12T08:23:15.546 DEBUG:teuthology.orchestra.run:got remote process result: 2
2019-11-12T08:23:15.549 INFO:teuthology.orchestra.run.smithi154.stderr:ls: cannot access /sys/fs/fuse/connections: No such file or directory
2019-11-12T08:23:15.550 DEBUG:teuthology.orchestra.run:got remote process result: 1
2019-11-12T08:23:15.550 ERROR:teuthology.run_tasks:Saw exception from tasks.
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/run_tasks.py", line 89, in run_tasks
    manager.__enter__()
  File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191111.211020/qa/tasks/ceph_fuse.py", line 136, in task
    mount.mount()
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191111.211020/qa/tasks/cephfs/fuse_mount.py", line 33, in mount
    return self._mount(mount_path, mount_fs_name)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191111.211020/qa/tasks/cephfs/fuse_mount.py", line 142, in _mount
    self.fuse_daemon.wait()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 162, in wait
    self._raise_for_status()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 184, in _raise_for_status
    node=self.hostname, label=self.label
CommandFailedError: Command failed on smithi154 with status 1: 'sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage daemon-helper kill ceph-fuse -f --name client.0 /home/ubuntu/cephtest/mnt.0'
````

From: http://pulpito.ceph.com/pdonnell-2019-11-11_23:46:50-fs-wip-pdonnell-testing-20191111.211020-distro-basic-smithi/4497670/

MDS Log:

````
2019-11-12T08:23:14.236+0000 7f6418101700  1 --2- [v2:172.21.15.141:6832/1662385508,v1:172.21.15.141:6833/1662385508] >>  conn(0x55c4241f8800 0x55c424e4a680 unknown :-1 s=NONE pgs=0 cs=0 l=0 rx=0 tx=0).accept
2019-11-12T08:23:14.236+0000 7f6417900700  1 --2- [v2:172.21.15.141:6832/1662385508,v1:172.21.15.141:6833/1662385508] >>  conn(0x55c4241f8800 0x55c424e4a680 unknown :-1 s=BANNER_ACCEPTING pgs=0 cs=0 l=0 rx=0 tx=0)._handle_peer_banner_payload supported=0 required=0
2019-11-12T08:23:14.237+0000 7f6417900700 10 mds.b parse_caps: parsing auth_cap_str='allow'
2019-11-12T08:23:14.237+0000 7f6417900700  1 --2- [v2:172.21.15.141:6832/1662385508,v1:172.21.15.141:6833/1662385508] >> 172.21.15.154:0/814922713 conn(0x55c4241f8800 0x55c424e4a680 crc :-1 s=READY pgs=4 cs=0 l=0 rx=0 tx=0).ready entity=client.4618 client_cookie=2258c5e35aed0325 server_cookie=1591f6926d5e95a0 in_seq=0 out_seq=0
2019-11-12T08:23:14.237+0000 7f64150fb700 10 mds.b  new session 0x55c424ea9800 for client.4618 172.21.15.154:0/814922713 con 0x55c4241f8800
2019-11-12T08:23:14.237+0000 7f64150fb700 10 mds.b parse_caps: parsing auth_cap_str='allow'
2019-11-12T08:23:14.237+0000 7f64150fb700 10 mds.b ms_handle_accept 172.21.15.154:0/814922713 con 0x55c4241f8800 session 0x55c424ea9800
2019-11-12T08:23:14.237+0000 7f64150fb700  1 -- [v2:172.21.15.141:6832/1662385508,v1:172.21.15.141:6833/1662385508] <== client.4618 172.21.15.154:0/814922713 1 ==== client_session(request_open) v3 ==== 329+0+0 (crc 0 0 0) 0x55c4241a0d80 con 0x55c4241f8800
2019-11-12T08:23:14.237+0000 7f64150fb700 20 mds.0.5 get_session have 0x55c424ea9800 client.4618 172.21.15.154:0/814922713 state closed
2019-11-12T08:23:14.237+0000 7f64150fb700  3 mds.0.server handle_client_session client_session(request_open) v3 from client.4618
2019-11-12T08:23:14.237+0000 7f64150fb700  0 mds.0.server  ignoring sessionless msg client_session(request_open) v3
2019-11-12T08:23:14.237+0000 7f64150fb700  1 -- [v2:172.21.15.141:6832/1662385508,v1:172.21.15.141:6833/1662385508] --> 172.21.15.154:0/814922713 -- client_session(reject) v3 -- 0x55c424db8b40 con 0x55c4241f8800
````

Reply:
> Get this failure:
> 
> ```
> 2019-11-12T08:23:14.232 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi154.stderr:2019-11-12T08:23:14.235+0000 7f0eb2b37e00 -1 init, newargv = 0x55cf51d09260 newargc=7
> 2019-11-12T08:23:14.233 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi154.stderr:ceph-fuse[17353]: starting ceph client
> 2019-11-12T08:23:14.236 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi154.stderr:2019-11-12T08:23:14.238+0000 7f0e9affd700 -1 client.4618 mds.0 rejected us (sessionless)
> 2019-11-12T08:23:14.236 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi154.stderr:ceph-fuse[17353]: ceph mount failed with (1) Operation not permitted
> 2019-11-12T08:23:14.244 DEBUG:teuthology.orchestra.run:got remote process result: 2
> 2019-11-12T08:23:14.255 INFO:teuthology.orchestra.run.smithi154.stderr:ls: cannot access /sys/fs/fuse/connections: No such file or directory
> 2019-11-12T08:23:14.402 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi154.stderr:daemon-helper: command failed with exit status 1
> 2019-11-12T08:23:15.357 INFO:teuthology.orchestra.run:Running command with timeout 900
> 2019-11-12T08:23:15.357 INFO:teuthology.orchestra.run.smithi154:Running:
> 2019-11-12T08:23:15.358 INFO:teuthology.orchestra.run.smithi154:> sudo mount -t fusectl /sys/fs/fuse/connections /sys/fs/fuse/connections
> 2019-11-12T08:23:15.477 DEBUG:teuthology.orchestra.run:got remote process result: 32
> 2019-11-12T08:23:15.478 INFO:teuthology.orchestra.run.smithi154.stderr:mount: mount point /sys/fs/fuse/connections does not exist
> 2019-11-12T08:23:15.479 INFO:teuthology.orchestra.run:Running command with timeout 900
> 2019-11-12T08:23:15.479 INFO:teuthology.orchestra.run.smithi154:Running:
> 2019-11-12T08:23:15.479 INFO:teuthology.orchestra.run.smithi154:> ls /sys/fs/fuse/connections
> 2019-11-12T08:23:15.546 DEBUG:teuthology.orchestra.run:got remote process result: 2
> 2019-11-12T08:23:15.549 INFO:teuthology.orchestra.run.smithi154.stderr:ls: cannot access /sys/fs/fuse/connections: No such file or directory
> 2019-11-12T08:23:15.550 DEBUG:teuthology.orchestra.run:got remote process result: 1
> 2019-11-12T08:23:15.550 ERROR:teuthology.run_tasks:Saw exception from tasks.
> Traceback (most recent call last):
>   File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/run_tasks.py", line 89, in run_tasks
>     manager.__enter__()
>   File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
>     return self.gen.next()
>   File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191111.211020/qa/tasks/ceph_fuse.py", line 136, in task
>     mount.mount()
>   File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191111.211020/qa/tasks/cephfs/fuse_mount.py", line 33, in mount
>     return self._mount(mount_path, mount_fs_name)
>   File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191111.211020/qa/tasks/cephfs/fuse_mount.py", line 142, in _mount
>     self.fuse_daemon.wait()
>   File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 162, in wait
>     self._raise_for_status()
>   File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 184, in _raise_for_status
>     node=self.hostname, label=self.label
> CommandFailedError: Command failed on smithi154 with status 1: 'sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage daemon-helper kill ceph-fuse -f --name client.0 /home/ubuntu/cephtest/mnt.0'
> ```
> 
> From: http://pulpito.ceph.com/pdonnell-2019-11-11_23:46:50-fs-wip-pdonnell-testing-20191111.211020-distro-basic-smithi/4497670/
> 
> MDS Log:
> 
> ```
> 2019-11-12T08:23:14.236+0000 7f6418101700  1 --2- [v2:172.21.15.141:6832/1662385508,v1:172.21.15.141:6833/1662385508] >>  conn(0x55c4241f8800 0x55c424e4a680 unknown :-1 s=NONE pgs=0 cs=0 l=0 rx=0 tx=0).accept
> 2019-11-12T08:23:14.236+0000 7f6417900700  1 --2- [v2:172.21.15.141:6832/1662385508,v1:172.21.15.141:6833/1662385508] >>  conn(0x55c4241f8800 0x55c424e4a680 unknown :-1 s=BANNER_ACCEPTING pgs=0 cs=0 l=0 rx=0 tx=0)._handle_peer_banner_payload supported=0 required=0
> 2019-11-12T08:23:14.237+0000 7f6417900700 10 mds.b parse_caps: parsing auth_cap_str='allow'
> 2019-11-12T08:23:14.237+0000 7f6417900700  1 --2- [v2:172.21.15.141:6832/1662385508,v1:172.21.15.141:6833/1662385508] >> 172.21.15.154:0/814922713 conn(0x55c4241f8800 0x55c424e4a680 crc :-1 s=READY pgs=4 cs=0 l=0 rx=0 tx=0).ready entity=client.4618 client_cookie=2258c5e35aed0325 server_cookie=1591f6926d5e95a0 in_seq=0 out_seq=0
> 2019-11-12T08:23:14.237+0000 7f64150fb700 10 mds.b  new session 0x55c424ea9800 for client.4618 172.21.15.154:0/814922713 con 0x55c4241f8800
> 2019-11-12T08:23:14.237+0000 7f64150fb700 10 mds.b parse_caps: parsing auth_cap_str='allow'
> 2019-11-12T08:23:14.237+0000 7f64150fb700 10 mds.b ms_handle_accept 172.21.15.154:0/814922713 con 0x55c4241f8800 session 0x55c424ea9800
> 2019-11-12T08:23:14.237+0000 7f64150fb700  1 -- [v2:172.21.15.141:6832/1662385508,v1:172.21.15.141:6833/1662385508] <== client.4618 172.21.15.154:0/814922713 1 ==== client_session(request_open) v3 ==== 329+0+0 (crc 0 0 0) 0x55c4241a0d80 con 0x55c4241f8800
> 2019-11-12T08:23:14.237+0000 7f64150fb700 20 mds.0.5 get_session have 0x55c424ea9800 client.4618 172.21.15.154:0/814922713 state closed
> 2019-11-12T08:23:14.237+0000 7f64150fb700  3 mds.0.server handle_client_session client_session(request_open) v3 from client.4618
> 2019-11-12T08:23:14.237+0000 7f64150fb700  0 mds.0.server  ignoring sessionless msg client_session(request_open) v3
> 2019-11-12T08:23:14.237+0000 7f64150fb700  1 -- [v2:172.21.15.141:6832/1662385508,v1:172.21.15.141:6833/1662385508] --> 172.21.15.154:0/814922713 -- client_session(reject) v3 -- 0x55c424db8b40 con 0x55c4241f8800
> ```

changed